pkg/credentials/aws/aws.go

```diff
@@ -221,6 +224,21 @@ func NewAuthEnvironment(region string, getter KeyGetter) Retriever {
 	}
 }
 
+// NewAuthProfile creates a retriever that fetches credentials from AWS profile
+// environment variable
+func NewAuthProfile(region string, getter KeyGetter) Retriever {
```

Comment:
What I meant when I suggested implementing this function was to make it do something like:

```go
// pseudo code
func NewAuthProfile(getter KeyGetter) Retriever {
    profile := getter("AWS_PROFILE")

    credentialsFile := ReadAll("well known credentials file path")

    credentials := make(map[string]Credentials)

    err := ini-library.Unmarshal(credentialsFile, &credentials)

    return &AuthStatic { // or other relevant
        Credentials: credentials[profile]
        IsValid: validate(credentials[profile])
    }
}
```

Doesn't that make sense for this use case?